include/boost/json/value_builder.hpp

```diff
@@ -0,0 +1,556 @@
+        array currently being built, which was created
+        by the last call to @ref on_array_begin, is finished.
+
+        @return `true` on success.
+
+        @param ec Set to the error, if any occurred.
+    */
+    BOOST_JSON_DECL
+    bool
+    on_array_end(
+        error_code& ec);
+
+    /** Continue creating a key.
+
+        This function appends the specified characters
+        to the key being built as the next element of
+        a currently open object. If a key is not currently
+        being built, the behavior is undefined.
+
+        @return `true` on success.
+
+        @param s The characters to append. This may be empty.
+
+        @param ec Set to the error, if any occurred.
+    */
+    BOOST_JSON_DECL
+    bool
+    on_key_part(
+        string_view s,
+        error_code& ec);
+
+    /** Finish creating a key.
+
+        This function appends the specified characters
+        to the key being built as the next element of
+        a currently open object, and finishes construction
+        of the key. If a key is not currently being built,
+        the behavior is undefined.
+
+        @return `true` on success.
+
+        @param s The characters to append. This may be empty.
+
+        @param ec Set to the error, if any occurred.
+    */
+    BOOST_JSON_DECL
+    bool
+    on_key(
+        string_view s,
+        error_code& ec);
+
+    /** Begin or continue creating a string.
+
+        This function appends the specified characters
+        to the string being built. If a string is not
+        currently being built, then a new empty string
+        is started.
+
+        @return `true` on success.
+
+        @param s The characters to append. This may be empty.
+
+        @param ec Set to the error, if any occurred.
+    */
+    BOOST_JSON_DECL
+    bool
+    on_string_part(
+        string_view s,
+        error_code& ec);
+
+    /** Create a string or finish creating a string.
+
+        This function appends the specified characters
+        to the string being built. If a string is not
+        currently being built, then a new string is created
+        with the specified characters.
+
+        @return `true` on success.
+
+        @param s The characters to append. This may be empty.
+
+        @param ec Set to the error, if any occurred.
+    */
+    BOOST_JSON_DECL
+    bool
+    on_string(
+        string_view s,
+        error_code& ec);
+
+    /** Begin building a number from a string.
+
+        This instructs the builder to begin building
+        a new JSON number, either as the top-level
+        element of the resulting value, or as the
+        next element of the current object or array
+        being built.
+
+        @note This function has no effect and always
+        returns `true`.
+
+        @return `true` on success.
+
+        @param s The characters to append. This may be empty.
+
+        @param ec Set to the error, if any occurred.
+    */
+    bool
+    on_number_part(
```

Comment:
Instead of manually casting to `void`, we can just leave the parameters unnamed.

Reply:
Then the docs won't be right